gufe/tests/test_protocol.py

```diff
@@ -696,7 +696,8 @@ def test_execute_DAG_retries(solvated_ligand, vacuum_ligand, tmpdir):
         # we did 3 retries, so 4 total failures
         assert len(r.protocol_unit_results) == 5
         assert len(r.protocol_unit_failures) == 4
-        assert len(list(shared.iterdir())) == 5
+        assert len(list(shared.iterdir())) == 0
+        # assert len(list(shared.iterdir())) == 5
```

Comment:
(Extremely minor) breaking change: Previously, this left behind 5 empty shared directories. In the current implementation, empty directories are, by default, automatically deleted, so we have no directories within the `shared` space. Had these directories contained any files, you would still find all 5.

There's a parameter to keep empty directories, but the effect will also be to keep empty subdirectories, which I didn't think was the desired default behavior. My assumption is that it will be relatively rare to keep `shared` anyway, because there should be something in settings to determine whether any given file is `scratch`/`shared`/`permanent`, and anything intended for long-term storage will be put in `permanent`.